src/CBT.NuGet/Internal/AggregatePackage.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading;
+
+namespace CBT.NuGet.Internal
+{
+    internal sealed class AggregatePackage
+    {
+        public enum AggregateOption
+        {
+            Add,
+            Remove,
+        }
+
+        public AggregatePackage(string outPropertyId, List<KeyValuePair<AggregateOption, string>> packagesToAggregate, string destinationRoot)
```

Comment:
I was just creating an ordered list of tuple pairs with <Action, Folder> action being add or remove from aggregate and folder being the folder that should be added to the aggregate or have it's contents removed from the aggregate.  

Instead of using Tuple I just used a KVP which technically isn't correct.  If I were to make the folder path a key then that could be unique but I didn't feel the need to remove duplicates as I didn't want to have to reason about which was the right one to remove.  And I didn't feel the need to throw an error on a duplicate entry as it would be a harmless authoring error that could be intentional, (though wasted time) due to prior constraints of the project.

Anyway let me know if you have a better object structure in mind.